plugin-server/src/utils/db/db.ts

```diff
@@ -1259,11 +1268,111 @@ export class DB {
+
+        if (options.forUpdate) {
+            queryString = queryString.concat(` FOR UPDATE`)
+        }
+
+        const selectResult: QueryResult = await this.postgresQuery(
+            queryString,
+            [teamId, groupTypeIndex, groupKey],
+            'fetchGroup',
+            client
+        )
+
+        if (selectResult.rows.length > 0) {
+            const rawGroup: RawGroup = selectResult.rows[0]
+            return {
+                ...rawGroup,
+                created_at: DateTime.fromISO(rawGroup.created_at).toUTC(),
+                version: Number(rawGroup.version || 0),
+            }
+        }
+    }
+
+    public async insertGroup(
+        teamId: TeamId,
+        groupTypeIndex: GroupTypeIndex,
+        groupKey: string,
+        groupProperties: Properties,
+        createdAt: DateTime,
+        propertiesLastUpdatedAt: PropertiesLastUpdatedAt,
+        propertiesLastOperation: PropertiesLastOperation,
+        version: number,
+        client?: PoolClient
+    ): Promise<void> {
+        const result = await this.postgresQuery(
+            `
+            INSERT INTO posthog_group (team_id, group_key, group_type_index, group_properties, created_at, properties_last_updated_at, properties_last_operation, version)
+            VALUES ($1, $2, $3, $4, $5, $6, $7, $8)
+            ON CONFLICT (team_id, group_key, group_type_index) DO NOTHING
+            RETURNING version
+            `,
+            [
+                teamId,
+                groupKey,
+                groupTypeIndex,
+                JSON.stringify(groupProperties),
+                createdAt.toISO(),
+                JSON.stringify(propertiesLastUpdatedAt),
+                JSON.stringify(propertiesLastOperation),
+                version,
+            ],
+            'upsertGroup',
+            client
+        )
+
+        if (result.rows.length === 0) {
+            throw new RaceConditionError('Parallel posthog_group inserts, retry')
+        }
+    }
+
+    public async updateGroup(
+        teamId: TeamId,
+        groupTypeIndex: GroupTypeIndex,
+        groupKey: string,
+        groupProperties: Properties,
+        createdAt: DateTime,
+        propertiesLastUpdatedAt: PropertiesLastUpdatedAt,
+        propertiesLastOperation: PropertiesLastOperation,
+        version: number,
+        client?: PoolClient
+    ): Promise<void> {
+        await this.postgresQuery(
+            `
+            UPDATE posthog_group
+            SET group_properties = $4, properties_last_updated_at = $5, properties_last_operation = $6, version = $7
+            WHERE team_id = $1 AND group_key = $2 AND group_type_index = $3
+            `,
+            [
+                teamId,
+                groupKey,
+                groupTypeIndex,
+                JSON.stringify(groupProperties),
+                JSON.stringify(propertiesLastUpdatedAt),
+                JSON.stringify(propertiesLastOperation),
+                version,
```

Comment:
I was wrong about your version usage initially because yes you used the DB value there via EXCLUDE, but the comment does apply now: https://github.com/PostHog/posthog/pull/7180/files/279551d4a2989665b8462663b9cc719d80330800#r751799192 
Plus I find it easier to read if we don't pass `version` to `insertGroup` function, which removing it from here will allow us to do.

Alternatively/Additionally should we throw/log if we couldn't do the update, i.e. the row wasn't locked as that might help us catch/debug bad situations faster